### PR TITLE
Promote existing e2e test for pod eviction on no-execute taint to conformance

### PR DESCRIPTION
**Pull request by globervinodhn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This PR requests for promoting an existing e2e test to conformance. The concerned e2e test verifies that the pod is evicted from the node once the node is hit with "no-execute" taint.

**Which issue(s) this PR fixes**: This PR requests for an existing e2e test to be promoted to conformance.
Fixes #

**Special notes for your reviewer**:
1. The concerned E2E test case executes in 69 to 75 seconds approximately.
2. The test is found to be non-flaky and non-disruptive in nature.

**Does this PR introduce a user-facing change?**:
NONE

release-note-none
/area conformance

cc @mgdevstack  @brahmaroutu

### CODE REVIEW

File: test/e2e/scheduling/taints.go [hunk 1]
```diff
@@ -166,10 +166,13 @@ var _ = SIGDescribe("NoExecuteTaintManager Single Pod [Serial]", func() {
...
 
-	// 1. Run a pod
-	// 2. Taint the node running this pod with a no-execute taint
-	// 3. See if pod will get evicted
-	It("evicts pods from tainted nodes", func() {
+	/*
+		Release : v1.15
+		Testname: Taint, Pod Eviction on no-execute
+		Description: Create a Pod and wait for it to be scheduled onto a Node. Taint the concerned Node
+						with a no-execute taint and wait for the Pod to be evicted. The test MUST pass.
```
Comment (spiffxp) [inline test/e2e/scheduling/taints.go]: please describe the behavior in terms of MUST language, not the test.. ie: the post MUST be evicted

Comment (spiffxp) [reply test/e2e/scheduling/taints.go]: If this test is worth promoting, why not the below test case that uses tolerations?

**Review by spiffxp — COMMENTED** [review]
/ok-to-test

Still need to review all the framework calls

**Comment by spiffxp** [comment]
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduling/taints.go#L131-L137 - it looks to me like something gets sent through the observed deletion channel regardless of whether the thing being deleted is a Pod

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduling/taints.go#L196 - the test doesn't bother to verify what came through the observed deletion channel, only that something does (maybe it should compare against `pod.Name`?)

I feel like these cases need to be tightened up, but I'm curious if others think I'm being overly scrupulous here

**Comment by bclau** [comment]
/test pull-kubernetes-e2e-gce-100-performance

FWIW, we're ok with this on the windows side.

File: test/e2e/scheduling/taints.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/taints.go]: I agree, this test also needs to have atleast 2 additional cases that I can think of

- Add a toleration for `No-Execute` taint and wait if it tolerates.
- Add `tolerationSeconds` with some value like `120` and check if we are waiting for those seconds before evicting.

@globervinodhn - Will you be able to update this PR with those changes?

Comment (spiffxp) [reply test/e2e/scheduling/taints.go]: That's a lot of time to wait for eviction. I'm not asking so much for writing more test cases, just that if we're promoting this test case, there appear to be more within this file that are eligible.

Comment (timothysc) [reply test/e2e/scheduling/taints.go]: https://github.com/kubernetes/kubernetes/pull/77331#issuecomment-491965511

**Comment by globervinodhn** [comment]
Including E2E test from #77614 and #77331 

All the tests belong and deal with TAINTS in kubernetes and can be promoted at once.

**Comment by oomichi** [comment]
I confirmed these tests have not been changed since https://github.com/kubernetes/kubernetes/commit/8726ed29eff2d05bbf2a694979df13d7de4d5333 at least except coding style changes (fixing golint failures, use changed test framework methods). That means I feel they are stable(not flake) to be promoted.

**Comment by spiffxp** [comment]
/ok-to-test
/priority important-soon
/milestone v1.15
I think this is worth allowing in during code freeze, there is no change in behavior, and the tests are already present in https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-serial&include-filter-by-regex=NoExecute

/approve
For the test/e2e changes
/lgtm

**Comment by bgrant0607** [comment]
@spiffxp @dims Did earlier concerns get addressed? I don't see comments about them being resolved.

**Comment by dims** [comment]
i'll let @spiffxp speak to that, all i did earlier was to punt this to next release. haven't looked deeper yet.

**Comment by spiffxp** [comment]
yeah ok upon further inspection my concerns from https://github.com/kubernetes/kubernetes/pull/76985#issuecomment-489828989 still stand... I feel like these tests need to be more explicit in what they're checking, and am open to opinions to the contrary

**Comment by johnbelamaric** [comment]
@globervinodhn this is waiting on https://github.com/kubernetes/kubernetes/pull/76985#issuecomment-500052076

**Comment by mgdevstack** [comment]
/hold 

until https://github.com/kubernetes/kubernetes/pull/76985#issuecomment-500052076 is addressed.

**Comment by spiffxp** [comment]
needs rebase

**Comment by globervinodhn** [comment]
Closed by mistake

**Comment by spiffxp** [comment]
@globervinodhn do you have any PRs in-flight to addres https://github.com/kubernetes/kubernetes/pull/76985#issuecomment-489828989 ? otherwise I feel like we should close this as we're not going to promote these tests in their current state

**Comment by spiffxp** [comment]
pinging for signs of life

**Comment by johnbelamaric** [comment]
I don't think we can promote these due to https://github.com/kubernetes/kubernetes/pull/76985#pullrequestreview-272768386

**Comment by xmudrii** [comment]
Due to inactivity and stale lifecycle, I'll remove this issue from the milestone. If there are any concerns, please ping me or SIG-Release on Slack.

/milestone clear

**Comment by johnbelamaric** [comment]
@jefftree PTAL

We have now concluded it is OK to promote disruptive tests if we mark them as such. Please check if these tests have been promoted by any other PR. If not, create a new PR to mark them as Disruptive and promote them. This PR can be closed (I don't think I can do it but I'll try...)

/close

**Pull request closed without merging**